src/nexus/utils.py

```diff
@@ -54,7 +59,8 @@ def get_total_records(self, endpoint: str) -> int:
     def get_paginated_resources(
             self,
             endpoint: str,
-            offset: typing.Optional[int] = 0
+            offset: typing.Optional[int] = 0,
+            params: dict = dict
```

Comment:
Same comment as above wrt using `dict` 

```suggestion
            params: typing.Optional[typing.Dict] = None
```

Reply:
updated @ricardogsilva